File: android/src/androidTest/java/org/ligi/passandroid/injections/FixedPassListPassStore.java
package org.ligi.passandroid.injections;

import android.support.annotation.Nullable;

import org.ligi.passandroid.model.FiledPass;
import org.ligi.passandroid.model.Pass;
import org.ligi.passandroid.model.PassClassifier;
import org.ligi.passandroid.model.PassStore;

import java.util.HashMap;
import java.util.List;

public class FixedPassListPassStore implements PassStore {

    private final List<FiledPass> passes;
    private Pass actPass;
    private PassClassifier passClassifier;

    public FixedPassListPassStore(List<FiledPass> passes) {
        this.passes = passes;
    }

    @Override
    public List<FiledPass> getPassList() {
        return passes;
    }

    @Override
    public void preCachePassesList() {
        // no effect in this impl
    }

    @Override
    public void deleteCacheForId(String id) {
        // no effect in this impl
    }

    @Override
    public void refreshPassesList() {
        // no effect in this impl
    }

    @Override
    @Nullable
    public Pass getPassbookForId(String id) {
        for (Pass pass : passes) {
            if (pass.getId().equals(id)) {
                return pass;
            }
        }
        return null;
    }

    @Override
    public Pass getCurrentPass() {
        return actPass;
    }

    @Override
    public void setCurrentPass(Pass pass) {
        actPass = pass;
    }

    @Override
    public PassClassifier getClassifier() {
        if (passClassifier == null) {
            passClassifier = new PassClassifier(new HashMap<String, String>(), this);
        }
        return passClassifier;
    }

    @Override
    public boolean deletePassWithId(String id) {
        return false;
    }

    @Override
    public String getPathForID(String id) {
        return null;
    }
}

File: android/src/androidTest/java/org/ligi/passandroid/injections/FixedPassListPassStore.kt
package org.ligi.passandroid.injections

import org.ligi.passandroid.model.PassClassifier
import org.ligi.passandroid.model.PassStore
import org.ligi.passandroid.model.pass.Pass
import java.io.File

class FixedPassListPassStore(private var passes: List<Pass>) : PassStore {

    override lateinit var classifier: PassClassifier

    init {
        classifier = PassClassifier(HashMap<String, String>(), this)
    }

    fun setList(newPasses: List<Pass>, newCurrentPass: Pass? = newPasses.firstOrNull()) {
        currentPass = newCurrentPass
        passes = newPasses
        passMap.clear()
        passMap.putAll(createHashMap())

        classifier = PassClassifier(HashMap<String, String>(), this)
    }

    override var currentPass: Pass? = null

    override val passMap: HashMap<String, Pass> by lazy {
        return@lazy createHashMap()
    }

    private fun createHashMap(): HashMap<String, Pass> {
        val hashMap = HashMap<String, Pass>()

        passes.forEach { hashMap.put(it.id, it) }
        return hashMap
    }

    override fun getPassbookForId(id: String): Pass? {
        return passMap[id]
    }


    override fun deletePassWithId(id: String): Boolean {
        return false
    }

    override fun getPathForID(id: String): File {
        return File("")
    }

    override fun save(pass: Pass) {
        // no effect in this impl
    }

    override fun notifyChange() {
        // no effect in this impl
    }

    override fun syncPassStoreWithClassifier(defaultTopic: String) {
        // no effect in this impl
    }

}
